registry/storage/shares.go

```diff
@@ -284,21 +284,18 @@ func specShareToStorageShare(share *types.SSVShare) *storageShare {
 func (s *sharesStorage) storageShareToSpecShare(share *storageShare) (*types.SSVShare, error) {
 	committee := make([]*spectypes.ShareMember, len(share.Committee))
 	for i, c := range share.Committee {
-		// TODO: (genesis) this is needed to fill in genesis committees
-		// opPubKey, ok := s.operators[c.OperatorID]
-		// if !ok {
-		// 	return nil, fmt.Errorf("operator not found: %d", c.OperatorID)
-		// }
-
 		committee[i] = &spectypes.ShareMember{
 			Signer:      c.OperatorID,
 			SharePubKey: c.PubKey,
 		}
 	}
 
+	var validatorPubKey spectypes.ValidatorPK
+	copy(validatorPubKey[:], share.ValidatorPubKey)
```

Comment:
I think this can work `validatorPubKey := spectypes.ValidatorPK(share.ValidatorPubKey)`

Reply:
If we ensure that share.ValidatorPubKey is 48 bytes long, then directly converting using `validatorPubKey := spectypes.ValidatorPK(share.ValidatorPubKey)` is safe.
However, if we don’t check for length and the length is not 48 it can lead to runtime error.
a safer approach is:
```go
var validatorPubKey spectypes.ValidatorPK
copy(validatorPubKey[:], share.ValidatorPubKey)
```

lmk wdyt

Reply:
well, if the validator pubkey is not 48 bytes then its a bug. in that case we could panic anyway, or at least return a critical error.. at least the error will surface, with the copy approach the key would not be valid and we won't know. So I guess its best to ensure the size anyway

Reply:
sure, added length check